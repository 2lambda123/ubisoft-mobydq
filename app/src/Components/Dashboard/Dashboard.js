import React from 'react';
import { styles } from './../../styles/baseStyles'
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import SimpleLineChart from '../../SimpleLineChart';
import SimpleTable from '../../SimpleTable';


class Dashboard extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        <div className={classes.appBarSpacer} />
        <Typography variant="display1" gutterBottom className={classes.chartContainer}>
          Some awesome Ux lalala
        </Typography>
      </React.Fragment>
     )
  }
}

export default withStyles(styles)(Dashboard);