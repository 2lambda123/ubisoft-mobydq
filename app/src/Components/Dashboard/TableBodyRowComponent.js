import React from 'react';
import {styles} from './../../styles/baseStyles';
import {withStyles} from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import DoneIcon from '@material-ui/icons/Done';
import ClearIcon from '@material-ui/icons/Clear';
import TableRow from '@material-ui/core/TableRow'


class TableBodyRowComponent extends React.Component
{
    constructor(props)
    {
        super(props);

        this.state = {
            rowData: this.props.rowData,
        }
    }

        isFlagActive(flag) {
        if(typeof(flag) === "boolean") {
            if (flag) {
                return <DoneIcon color="primary"/>;
            }
            return <ClearIcon color="error"/>
        }
    }


    render()
    {
        let isFlagActiveFunction = this.isFlagActive;
        return(
            <TableRow>
                {this.state.rowData.map((value) => (
                    <TableCell key={value}>{value} {isFlagActiveFunction(value)}</TableCell>
                ))}
            </TableRow>
        )
    }
}

export default withStyles(styles)(TableBodyRowComponent);