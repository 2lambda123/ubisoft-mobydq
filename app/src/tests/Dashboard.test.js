import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Dashboard from './../Components/Dashboard/Dashboard';

it('renders empty dashboard crashing', () => {
  const wrapper = shallow(
    <Dashboard />
  );

  expect(wrapper).toMatchSnapshot();
});
