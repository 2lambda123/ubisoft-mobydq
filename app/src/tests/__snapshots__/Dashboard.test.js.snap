// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty dashboard crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Dashboard) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "appBar": "Dashboard-appBar-4",
        "appBarShift": "Dashboard-appBarShift-5",
        "appBarSpacer": "Dashboard-appBarSpacer-11",
        "chartContainer": "Dashboard-chartContainer-13",
        "content": "Dashboard-content-12",
        "drawerPaper": "Dashboard-drawerPaper-9",
        "drawerPaperClose": "Dashboard-drawerPaperClose-10",
        "formContainer": "Dashboard-formContainer-14",
        "menuButton": "Dashboard-menuButton-6",
        "menuButtonHidden": "Dashboard-menuButtonHidden-7",
        "root": "Dashboard-root-1",
        "tableContainer": "Dashboard-tableContainer-15",
        "title": "Dashboard-title-8",
        "toolbar": "Dashboard-toolbar-2",
        "toolbarIcon": "Dashboard-toolbarIcon-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "appBar": "Dashboard-appBar-4",
          "appBarShift": "Dashboard-appBarShift-5",
          "appBarSpacer": "Dashboard-appBarSpacer-11",
          "chartContainer": "Dashboard-chartContainer-13",
          "content": "Dashboard-content-12",
          "drawerPaper": "Dashboard-drawerPaper-9",
          "drawerPaperClose": "Dashboard-drawerPaperClose-10",
          "formContainer": "Dashboard-formContainer-14",
          "menuButton": "Dashboard-menuButton-6",
          "menuButtonHidden": "Dashboard-menuButtonHidden-7",
          "root": "Dashboard-root-1",
          "tableContainer": "Dashboard-tableContainer-15",
          "title": "Dashboard-title-8",
          "toolbar": "Dashboard-toolbar-2",
          "toolbarIcon": "Dashboard-toolbarIcon-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
